csi/moac/volume.ts

```diff
@@ -341,6 +341,19 @@ export class Volume {
         }
         return;
       }
+    } else {
+      if (this.nexus) {
+        if (this.nexus.node?.name !== this.publishedOn) {
+          // Respawn the nexus on the desired node.
+          log.info(`Recreating the nexus "${this.nexus}" on the desired node "${this.publishedOn}"`);
```

Comment:
So we destroy it now and then the fsa will recreate it on the correct node?

Reply:
Correct